src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java

```diff
@@ -500,6 +500,44 @@ public ValidationOutputData deleteKruizeExperimentEntryByName(String experimentN
         return validationOutputData;
     }
 
+    /**
+     * Delete metadata with the name dataSourceName
+     * This deletes the metadata from the KruizeDSMetadataEntry table
+     * @param dataSourceName
+     * @return
+     */
+    @Override
+    public ValidationOutputData deleteKruizeDSMetadataEntryByName(String dataSourceName) {
+        ValidationOutputData validationOutputData = new ValidationOutputData(false, null, null);
+        Transaction tx = null;
+        try (Session session = KruizeHibernateUtil.getSessionFactory().openSession()) {
+            try {
+                tx = session.beginTransaction();
+                Query query = session.createQuery(DELETE_FROM_METADATA_BY_DATASOURCE_NAME, null);
+                query.setParameter("dataSourceName", dataSourceName);
+                int deletedCount = query.executeUpdate();
+
+                if (deletedCount == 0) {
+                    validationOutputData.setSuccess(false);
```

Comment:
The condition should remain true unless an exception occurs, in which case it should be set to false. However, if the deleted count equals zero, the message should indicate either "No records found" or "Records already deleted."

Reply:
Please set up code walk through session